src/reason_parser.mly

```diff
@@ -3553,7 +3552,7 @@ constructor_declarations:
 ;
 
 constructor_declaration_no_leading_bar:
-  | as_loc(constr_ident) generalized_constructor_arguments attributes
+  | as_loc(UIDENT) generalized_constructor_arguments attributes
```

Comment:
Actually, instead of splitting `constr_ident` into two parts (UIDENT + non-UIDENT), there is another trick that I just used recently, which is to use `%inline`on constr_ident like:

```
%inline constr_ident:
    UIDENT                                      { $1 }
  | LPAREN RPAREN                               { "()" }
  | COLONCOLON                                  { "::" }
  | FALSE                                       { "false" }
  | TRUE                                        { "true" }
;
```

This trick basically expands into what you are manually writing now. For more information, take a look at:
[menhir's manual](http://gallium.inria.fr/~fpottier/menhir/manual.pdf), chapter 5.3